src/js/components/RadioButton/RadioButton.js

```diff
@@ -85,8 +102,18 @@ const RadioButton = forwardRef(
           <StyledRadioButtonInput
             aria-label={a11yTitle}
             {...rest}
-            ref={ref}
+            ref={(node) => {
+              inputRef.current = node;
+
+              const currentRef = ref;
+              if (typeof currentRef === 'function') {
+                currentRef(node);
+              } else if (currentRef && 'current' in currentRef) {
+                currentRef.current = node;
+              }
+            }}
```

Comment:
Is this basically trying to use the forwarded ref? Ie. I think you should just be using `useForwardedRef()` from utils